enhancements/api-review/add-mirror-by-digest-only.md

```diff
@@ -0,0 +1,226 @@
+type Registry struct {
+  // ...
+
+	// If true, mirrors will only be used for digest pulls. Pulling images by
+	// tag can potentially yield different images, depending on which endpoint
+	// we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+	MirrorByDigestOnly bool `toml:"mirror-by-digest-only,omitempty"`
+}
+```
+
+The ImageContentSourcePolicy CRD has the cluster-wide information about handle the mirrors only when pulling the images that are referenced by their digests.
+It will now watch for `mirror-by-digest-only` to not require digest reference.
+
+An example ImageContentSourcePolicy file will look like:
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+    mirror-by-digest-only: false
+```
+
+### User Stories
+
+#### As a user, I would like to pull images from mirror using tag, without digests reference
+The user can specify `mirror-by-digest-only: false` for the `source` under `repositoryDigestMirrors` in the ImageContentSourcePolicy file. And create the ImageContentSourcePolicy project. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+- openshift/api
+- openshift/machine-config-operator
+
+This is an example of the ImageContentSourcePolicy file:
+
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - example.io/example/ubi-minimal 
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+    mirror-by-digest-only: false 
+```
+
+The ImageContentSourcePolicy file will create a drop-in file at `/etc/containers/registries.conf` that looks like:
+
+```toml
+unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+[[registry]]
+  location = "registry.access.redhat.com/ubi8/"
+  insecure = false
+  blocked = false
+  # No mirror-by-digest-only = true configured
+  prefix = ""
+
+  [[registry.mirror]]
+    location = "example.io/example/ubi8-minimal"
+    insecure = false
+```
+
+If different mirror-by-digest-only values set for same “source” repository, the ImageContentSourcePolicy will be rejected.
+
+### Risks and Mitigations
+
+To pull images from mirror registries without the digest specifications, set the MirrorByDigestOnly false,
+this could lead to returning different image from different registry if the image tag mapping is out of sync.
+
+## Design Details
+
+### Test Plan
+
+Update the tests that are currently in the MCO to verify that `mirror-by-digest-only` not have been set when the ImageContentSourcePolicy is created to drop it.
+
+**Note:** *Section not required until targeted at a release.*
```

Comment:
This and later in this section look like template boilerplate that can be dropped.